pkg/controllers/provisioning/scheduling/scheduler.go

```diff
@@ -74,53 +74,64 @@ func (s *Scheduler) Solve(ctx context.Context, provisioner *v1alpha5.Provisioner
 	if err := s.Topology.Inject(ctx, constraints, pods); err != nil {
 		return nil, fmt.Errorf("injecting topology, %w", err)
 	}
-	// Separate pods into schedules of isomorphic scheduling constraints.
-	schedules, err := s.getSchedules(ctx, constraints, pods)
+	instanceTypes, err := s.CloudProvider.GetInstanceTypes(ctx, constraints.Provider)
```

Comment:
We're already getting instance types in provisioner.go. Can we pass them in rather than wiring up the cloudprovider in here?

Reply:
Yeah we definitely can. Done.